### PR TITLE
Refactored code to automatically map new ontologies

### LINKED ISSUE #304: Incorporate rejected mappings in pipeline
1. Include a `make` goal that clones the rejected mappings from @sabrinatoro TSV in `../mappings/rejected.sssom.tsv`.
2. Incorporated rejected mappings  by removing all matched rows where subject_id, predicate_id, object_id (nothing else) are in the rejected mappings

@hrshdhgd says there is already a function for that

### PR DESCRIPTION
**Pull request by hrshdhgd**
- [x] Refactored code to automatically map new ontologies
 - [x] Filters out rejected mappings : Fixes #304
 - [x] Deletes prior mappings before recreating them.

### CODE REVIEW

File: src/ontology/lexmatch/README.md [hunk 1]
```diff
@@ -3,49 +3,48 @@
 * mondo-only: Positive mappings in MONDO not caught by the lexical mapping pipeline
 * split-mapping-set: Unmapped mappings broken down by predicate_id
 ## Summary of mappings:
-## unmapped_xxxx_lex & unmapped_xxxx_lex_exact
- * Number of mappings in [`unmapped_icd_lex`](unmapped_icd_lex.tsv): 1928
```
Comment (matentzn) [inline src/ontology/lexmatch/README.md:7]: Can you sort the output on this file here so it does not get such a big diff each time? Its super useful when reviewing mondo-ingest runs to see the changed numbers!

File: src/ontology/mondo-ingest.Makefile [hunk 2]
```diff
@@ -432,9 +432,9 @@ lexical-matches: ../mappings/mondo-sources-all-lexical.sssom.tsv
 ###################################
 #### Lexmatch-SSSOM-compare #######
 ###################################
-lexmatch/README.md: $(SCRIPTSDIR)/lexmatch-sssom-compare.py ../mappings/mondo-sources-all-lexical.sssom.tsv
+lexmatch/README.md: $(SCRIPTSDIR)/lexmatch-sssom-compare.py
```
Comment (matentzn) [inline src/ontology/mondo-ingest.Makefile:435]: `../mappings/mondo-sources-all-lexical.sssom.tsv` should be a dependency here!

File: src/ontology/rejected-mappings.sssom.tsv
Comment (matentzn) [inline src/ontology/rejected-mappings.sssom.tsv:1]: move to src/mappings

File: src/ontology/rejected-mappings.sssom.tsv [hunk 3]
```diff
@@ -0,0 +1,26 @@
+subject_id	subject_label	object_id	predicate_id	object_label	mapping_justification	mapping_tool	confidence	subject_match_field	object_match_field	match_string	comment	predicate_modifier	reviewer_id	reason for rejected
+MONDO:0005571	polycythemia	NCIT:C27794	MONDO:equivalentTo	Polycythemia (Excluding Polycythemia Vera)	semapv:LexicalMatching	oaklib	0.8497788952	rdfs:label	rdfs:label	polycythemia	LEXMATCH	NOT	https://orcid.org/0000-0002-4142-7153	mapping rule incorrect
```
Comment (matentzn) [inline src/ontology/rejected-mappings.sssom.tsv:2]: Invalid `predicate_id`, see #333; having this correct is important for the code here to work, as we only reject mappings with the exact same predicate_id!

File: src/scripts/lexmatch-sssom-compare.py [hunk 4]
```diff
@@ -159,134 +144,49 @@ def extract_unmapped_matches(matches: TextIO, output_dir: str, summary: TextIO):
...
+    condition_lex_df_mondo_subj = lex_df["subject_id"].str.contains("MONDO")
+    ont_df_list = []
+
+    for idx, ont in enumerate(input):
+        ont2 = ont.upper()
+        if ont == "omim":
+            ont2 = "|".join((["OMIM", "OMIMPS"]))
+
+        mondo_ont_df = msdf_mondo.df[condition_mondo_sssom_subj & msdf_mondo.df['object_id'].str.contains(ont2)]
+        mondo_ont_lex_df = lex_df[(condition_lex_df_mondo_subj & lex_df['object_id'].str.contains(ont2))]
```
Comment (matentzn) [inline src/scripts/lexmatch-sssom-compare.py:156]: `contains` deals with this OMIM|OMIMP string above?

**Review by matentzn — COMMENTED** [review]
Few comments, looks awesome!

File: src/ontology/rejected-mappings.sssom.tsv
Comment (hrshdhgd) [reply src/ontology/rejected-mappings.sssom.tsv:1]: Done

File: src/scripts/lexmatch-sssom-compare.py (hunk 4 above)
Comment (hrshdhgd) [reply src/scripts/lexmatch-sssom-compare.py:156]: Yes

File: src/ontology/mondo-ingest.Makefile [hunk 5]
```diff
@@ -422,19 +426,27 @@ tmp/merged.db: tmp/merged.owl
 	@rm -f .template.db.tmp
 	@rm -f tmp/merged-relation-graph.tsv.gz
 
-../mappings/mondo-sources-all-lexical.sssom.tsv: $(SCRIPTSDIR)/match-mondo-sources-all-lexical.py tmp/merged.db
-	rm -f ../mappings/mondo-sources-all-lexical.sssom.tsv
-	rm -f ../mappings/mondo-sources-all-lexical-2.sssom.tsv
-	python $< run tmp/merged.db -c metadata/mondo.sssom.config.yml -r config/mondo-match-rules.yaml -o $@
+$(MAPPINGSDIR)/mondo-sources-all-lexical.sssom.tsv: $(SCRIPTSDIR)/match-mondo-sources-all-lexical.py tmp/merged.db $(MAPPINGSDIR)/"rejected-mappings.sssom.tsv"
```
Comment (matentzn) [inline src/ontology/mondo-ingest.Makefile:429]:
```suggestion
$(MAPPINGSDIR)/mondo-sources-all-lexical.sssom.tsv: $(SCRIPTSDIR)/match-mondo-sources-all-lexical.py tmp/merged.db $(MAPPINGSDIR)/rejected-mappings.sssom.tsv
```

**Review by matentzn — CHANGES_REQUESTED** [review]
One small fix to be applied, then Looks good

**Review by matentzn — APPROVED** [review]
Looks amazing!

File: src/ontology/mondo-ingest.Makefile [hunk 6]
```diff
@@ -422,19 +426,27 @@ tmp/merged.db: tmp/merged.owl
 	@rm -f .template.db.tmp
 	@rm -f tmp/merged-relation-graph.tsv.gz
 
-../mappings/mondo-sources-all-lexical.sssom.tsv: $(SCRIPTSDIR)/match-mondo-sources-all-lexical.py tmp/merged.db
-	rm -f ../mappings/mondo-sources-all-lexical.sssom.tsv
-	rm -f ../mappings/mondo-sources-all-lexical-2.sssom.tsv
-	python $< run tmp/merged.db -c metadata/mondo.sssom.config.yml -r config/mondo-match-rules.yaml -o $@
+$(MAPPINGSDIR)/mondo-sources-all-lexical.sssom.tsv: $(SCRIPTSDIR)/match-mondo-sources-all-lexical.py tmp/merged.db $(MAPPINGSDIR)/rejected-mappings.sssom.tsv
```
Comment (joeflack4) [inline src/ontology/mondo-ingest.Makefile:429]: Definitely looks a lot more succinct! Pretty big refactor in this PR.
I see a lot of usage of `ALL_COMPONENT_IDS`. My only comment is that the `mondo-sources-*-lexical*.sssom` outputs are coming out with all ontologies in 1 file without a previous step of 1 file per ontology. I don't have any qualms with that though so long as everything is working as intended!

Comment (hrshdhgd) [reply src/ontology/mondo-ingest.Makefile:429]: The split happens down the pipeline.

**Pull request merged**